go/libraries/events/event_flush.go

```diff
@@ -64,63 +75,167 @@ func getGRPCEmitter(dEnv *env.DoltEnv) *GrpcEmitter {
 	return NewGrpcEmitter(conn)
 }
 
-// NewEventGrpcFlush creates a new EventGrpcFlush
-func NewEventGrpcFlush(fs filesys.Filesys, userHomeDir string, doltDir string, dEnv *env.DoltEnv) *EventGrpcFlush {
+// lockAndFlush locks the given lockPath and passes the flushCB to the filesys' Iter method
+func lockAndFlush(ctx context.Context, fs filesys.Filesys, dirPath string, lockPath string, fcb flushCB) error {
+	fsLock := filesys.CreateFilesysLock(fs, lockPath)
+
+	isUnlocked, err := fsLock.TryLock()
+
+	defer func() error {
+		err := fsLock.Unlock()
+		if err != nil {
+			return err
+		}
+		return nil
+	}()
+
+	if !isUnlocked && err != nil {
+		if err == fslock.ErrLocked {
+			return ErrFileLocked
+		}
+		return err
+	}
+
+	if isUnlocked && err == nil {
+		err := fs.Iter(dirPath, false, func(path string, size int64, isDir bool) (stop bool) {
+			if err := fcb(ctx, path); err != nil {
+				// log.Print(err)
+				return false
+			}
+			return false
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GrpcEventFlusher parses dolt event logs sends the events to the events server
+type GrpcEventFlusher struct {
+	em       *GrpcEmitter
+	fbp      *FileBackedProc
+	LockPath string
+}
 
+// NewGrpcEventFlusher creates a new GrpcEventFlusher
+func NewGrpcEventFlusher(fs filesys.Filesys, userHomeDir string, doltDir string, dEnv *env.DoltEnv) *GrpcEventFlusher {
 	fbp := NewFileBackedProc(fs, userHomeDir, doltDir, MD5FileNamer, CheckFilenameMD5)
 
 	if exists := fbp.EventsDirExists(); !exists {
 		panic(ErrEventsDataDir)
 	}
 
-	return &EventGrpcFlush{em: getGRPCEmitter(dEnv), fbp: fbp}
+	return &GrpcEventFlusher{em: getGRPCEmitter(dEnv), fbp: fbp, LockPath: fbp.GetEventsDirPath()}
 }
 
-// flush is the FSIterCB function that sends the events requests from the .devts files to the grpc server
-func (egf *EventGrpcFlush) flush(path string, size int64, isDir bool) (stop bool) {
+// flush has the function signature of the flushCb type
+// and sends events data to the events server
+func (egf *GrpcEventFlusher) flush(ctx context.Context, path string) error {
 	fs := egf.fbp.GetFileSys()
 
 	data, err := fs.ReadFile(path)
 	if err != nil {
-		log.Print(err)
-		return false
+		return err
 	}
 
 	isFileValid, err := egf.fbp.CheckingFunc(data, path)
 
 	if isFileValid && err == nil {
-		ctx, cnclFn := context.WithDeadline(context.Background(), time.Now().Add(time.Minute))
```

Comment:
I believe the context already has a deadline set in the caller.  Probably can skip and just use the ctx object